src/odemis/driver/hamamatsurx.py

```diff
@@ -447,13 +444,24 @@ def _getDataFromBuffer(self):
                     try:
                         event_time = self.queue_events.popleft()
                         logging.warning("Starting acquisition delayed by %g s.", time.time() - event_time)
-                        self.parent.AcqStart(self.acqMode)
+                        self.parent.AcqStart("SingleLive")  # should never be a different
                         is_receiving_image = True
                     except IndexError:
                         # No event (yet) => fine
                         pass
 
-                rargs = self.parent.queue_img.get(block=True)  # block until receive something
+                if self._sync_event:
+                    timeout = self.exposureTime.value * 2
+                else:
+                    timeout = None
+
+                try:
+                    rargs = self.parent.queue_img.get(block=True, timeout=timeout)  # block until receive something
+                except queue.Empty:
+                    logging.warning("Failed to receive image from streak ccd (timed out). Will try again.")
```

Comment:
For debugging purpose, it might be handy to know how long it waited. Please add the `tiemout` value to the message.